controllers/cluster_controller.go

```diff
@@ -84,7 +83,6 @@ func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager, options controlle
 
 func (r *ClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr error) {
 	ctx := context.Background()
-	_ = r.Log.WithValues("cluster", req.NamespacedName)
```

Comment:
I'd recommend defining the log here and using this instead of `r.Log` in the function. This gives every log some nice context

Reply:
then we need to pass `logger` to every functions that log things :confused: 
another way to do is what capa did, a scope object is defined with a field named `logger`, and it is passed down to every funtion

Reply:
sorry, I meant for just this function.

Define the logger up front with the desired context (if applicable) and use it throughout the function. However, functions not defined on the reconciler *should* have a logger passed into them if they need one. Logging is a dependency and should be treated as such.

Reply:
oh, I got it :) thanks!

Reply:
can you do a quick review again?

Reply:
this is looking great!